### ISSUE TITLE
Qt detection broken for /cvmfs/icecube.opensciencegrid.org/py2-v1 (Trac #1330)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1322]
This clause in config.cmake:

```text
#
# Find CVMFS software
# (designed for py2-v2 port-less software)
#
if($ENV{SROOT} MATCHES "^/cvmfs/icecube" AND NOT SYSTEM_PACKAGES)
  set(CMAKE_PREFIX_PATH $ENV{SROOT})
  set(SYSTEM_PACKAGES TRUE)
  set(USE_CVMFS TRUE CACHE BOOL "Are we using CVMFS?")
endif()
```

unconditionally sets `SYSTEM_PACKAGES=TRUE` when using a cvmfs toolset. If the system is stock RHEL6 with Qt 4.6 installed, this will cause CMake to find the `/usr/bin/qt4-qmake` associated with 4.6 rather than the 4.8 shipped in `I3_PORTS`.

Do we really want to fall back to system search paths in this case? If so, then qt4.cmake needs to be taught how to select a minimum version of Qt rather than failing if the first one it finds is too old.

Migrated from https://code.icecube.wisc.edu/ticket/1330
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:01:05", 
    "description": "This clause in config.cmake:\n{{{\n#\n# Find CVMFS software\n# (designed for py2-v2 port-less software)\n#\nif($ENV{SROOT} MATCHES \"^/cvmfs/icecube\" AND NOT SYSTEM_PACKAGES)\n  set(CMAKE_PREFIX_PATH $ENV{SROOT})\n  set(SYSTEM_PACKAGES TRUE)\n  set(USE_CVMFS TRUE CACHE BOOL \"Are we using CVMFS?\")\nendif()\n}}}\n\nunconditionally sets `SYSTEM_PACKAGES=TRUE` when using a cvmfs toolset. If the system is stock RHEL6 with Qt 4.6 installed, this will cause CMake to find the `/usr/bin/qt4-qmake` associated with 4.6 rather than the 4.8 shipped in `I3_PORTS`.\n\nDo we really want to fall back to system search paths in this case? If so, then qt4.cmake needs to be taught how to select a minimum version of Qt rather than failing if the first one it finds is too old.", 
    "reporter": "jvansanten", 
    "cc": "hbretz", 
    "resolution": "fixed", 
    "_ts": "1547251265192575", 
    "component": "cmake", 
    "summary": "Qt detection broken for /cvmfs/icecube.opensciencegrid.org/py2-v1", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-09-04T19:46:59", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150905T17:49:46: 
* david.schultz commented:

> We set the `CMAKE_PREFIX_PATH` to make it prefer CVMFS over the base system.  Of course, our Qt detection could just be messed up.  I guess I can set up a VM to test this, since the cobalts don't have system qt.


* david.schultz changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20150907T12:22:25: 
* jvansanten commented:

> In changeset 2295:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2295"
> Test qmake version before proceeding with Qt detection. Fixes #1322.
> ```


* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "assigned" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:02: 
* olivas changed milestone from "September 2015" to ""
* olivas commented:

> Milestone September 2015 deleted

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:01:05: nwhitehorn commented:

> In changeset 170113:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170113"
> Test qmake version before proceeding with Qt detection. Fixes #1322.
> ```
